### PR TITLE
Various path/type cleanups. [MDB IGNORE] [IDB IGNORE]

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
Sorts out out the various exception types in #2575 as best as possible.
Implements abstract type checking on /datum and adds a var to prevent manual/unit test spawning of types.

## Why and what will this PR improve
Safer init, more systematic handling of abstract objects.

## Authorship
Myself.

## Changelog
Nothing player-facing.

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Has dependencies:
~~#2726~~
~~#2720~~
~~#2727~~
~~#2713~~

File: code/unit_tests/items.dm [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		volatile_tests += new test_path(src)
+
+	//Run tests on each objects
+	for(var/objpath in obj_test_instances)
+		//Run constant tests first
+		for(var/datum/item_unit_test/T in constant_tests)
+			T.run_test(obj_test_instances[objpath])
+		//Run volatile tests second
+		for(var/datum/item_unit_test/T in volatile_tests)
+			T.run_test(obj_test_instances[objpath])
```
Comment (afterthought2) [inline code/unit_tests/items.dm]: Probably should at least have them check the return value for pass/fail, even if logging happens in the test itself. Also, are there tests being added with this system?

File: code/modules/materials/materials_ore.dm [hunk 2]
```diff
@@ -113,49 +114,70 @@
...
+	is_spawnable_type = TRUE
 /obj/item/stack/material/ore/neon
 	material = /decl/material/solid/ice/hydrate/neon
+	is_spawnable_type = TRUE
 /obj/item/stack/material/ore/krypton
 	material = /decl/material/solid/ice/hydrate/krypton
+	is_spawnable_type = TRUE
 /obj/item/stack/material/ore/xenon
 	material = /decl/material/solid/ice/hydrate/xenon
+	is_spawnable_type = TRUE
```
Comment (afterthought2) [inline code/modules/materials/materials_ore.dm]: You can't do this at /ore level or under another common type?

File: code/unit_tests/items.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/unit_tests/items.dm]: No, I'm only interested in whether or not all the types init without runtimes.

File: code/modules/materials/materials_ore.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/modules/materials/materials_ore.dm]: Now that I check it looks like `/obj/item/stack/material/ore` has a default material, so should be fine to move it down to that.

File: code/unit_tests/items.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/unit_tests/items.dm]: Happy to leave the individual test failure stuff to Psy_Commando's PR as it's not really needed here.

Comment (afterthought2) [reply code/unit_tests/items.dm]: If the idea is that the mini-tests modify the big test's `failures` list themselves (not how I'd do it but sure, fine), document that expectation below.

Comment (MistakeNot4892) [reply code/unit_tests/items.dm]: Made failure reporting explicit.

**Review by afterthought2 — APPROVED** [review]

**Pull request merged**